pkg/controller/eks/cluster.go

```diff
@@ -0,0 +1,240 @@
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(resource.Ignore(eks.IsErrorNotFound, err), errDescribeFailed)
+	}
+
+	current := cr.Spec.ForProvider.DeepCopy()
+	eks.LateInitialize(&cr.Spec.ForProvider, rsp.Cluster)
+	if !reflect.DeepEqual(current, &cr.Spec.ForProvider) {
+		if err := e.kube.Update(ctx, cr); err != nil {
+			return managed.ExternalObservation{}, errors.Wrap(err, errKubeUpdateFailed)
+		}
+	}
+
+	cr.Status.AtProvider = eks.GenerateObservation(rsp.Cluster)
+	switch cr.Status.AtProvider.Status {
+	case v1beta1.ClusterStatusActive:
+		cr.Status.SetConditions(runtimev1alpha1.Available())
+		resource.SetBindable(cr)
+	case v1beta1.ClusterStatusCreating:
+		cr.Status.SetConditions(runtimev1alpha1.Creating())
+	case v1beta1.ClusterStatusDeleting:
+		cr.Status.SetConditions(runtimev1alpha1.Deleting())
+	default:
+		cr.Status.SetConditions(runtimev1alpha1.Unavailable())
+	}
+	upToDate, err := eks.IsUpToDate(&cr.Spec.ForProvider, rsp.Cluster)
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(err, errUpToDateFailed)
+	}
+
+	return managed.ExternalObservation{
+		ResourceExists:    true,
+		ResourceUpToDate:  upToDate,
+		ConnectionDetails: eks.GetConnectionDetails(rsp.Cluster, e.sts),
+	}, nil
+}
+
+func (e *external) Create(ctx context.Context, mg resource.Managed) (managed.ExternalCreation, error) {
+	cr, ok := mg.(*v1beta1.Cluster)
+	if !ok {
+		return managed.ExternalCreation{}, errors.New(errNotEKSCluster)
+	}
+	cr.SetConditions(runtimev1alpha1.Creating())
+	if cr.Status.AtProvider.Status == v1beta1.ClusterStatusCreating {
+		return managed.ExternalCreation{}, nil
+	}
+	_, err := e.client.CreateClusterRequest(eks.GenerateCreateClusterInput(meta.GetExternalName(cr), &cr.Spec.ForProvider)).Send(ctx)
+	return managed.ExternalCreation{}, errors.Wrap(err, errCreateFailed)
+}
+
+func (e *external) Update(ctx context.Context, mg resource.Managed) (managed.ExternalUpdate, error) {
+	cr, ok := mg.(*v1beta1.Cluster)
+	if !ok {
+		return managed.ExternalUpdate{}, errors.New(errNotEKSCluster)
+	}
+	switch cr.Status.AtProvider.Status {
+	case v1beta1.ClusterStatusUpdating, v1beta1.ClusterStatusCreating:
+		return managed.ExternalUpdate{}, nil
+	}
+
+	// NOTE(hasheddan): we have to describe the cluster again because different
+	// fields require different update methods.
+	rsp, err := e.client.DescribeClusterRequest(&awseks.DescribeClusterInput{Name: aws.String(meta.GetExternalName(cr))}).Send(ctx)
+	if err != nil {
+		return managed.ExternalUpdate{}, errors.Wrap(err, errDescribeFailed)
+	}
+	patch, err := eks.CreatePatch(rsp.Cluster, &cr.Spec.ForProvider)
+	if err != nil {
+		return managed.ExternalUpdate{}, errors.Wrap(err, errPatchCreationFailed)
+	}
+	if _, err = e.client.UpdateClusterConfigRequest(eks.GenerateUpdateClusterConfigInput(meta.GetExternalName(cr), patch)).Send(ctx); err != nil {
+		return managed.ExternalUpdate{}, errors.Wrap(resource.Ignore(eks.IsErrorInUse, err), errUpdateConfigFailed)
```

Comment:
Note that I am ignoring `IsErrorInUse` so that we do a full wait if another Update operation is still underway. This means that when this error occurs we will report in logs / status that a successful Update request was made. I think this is fine as the only time this error will occur is when we are already updating some other field, and we will eventually finish that update and then try this one again.

Reply:
If these requests cause the state to go into `Updating` state, then it seems that should be fine after the next reconcile. However, I'm wondering whether an empty `UpdateClusterConfigRequest` will cause that as well, which might render the rest of the logic useless.

Reply:
@muvaf I can test again, but given that I was able to update the cluster version, which is a separate call after this one, I am guessing that an empty update is just ignored